### PR TITLE
Adding a MaximumRequestSizeHandler

### LINKED ISSUE #1143: Have maximum request size by default
It would be great to check incoming requests to make sure they are below a certain size. Otherwise, a malicious client could hit the server with a huge request and cause a denial-of-service attack. Perhaps this could be an adjustable value in `config/server.cr`.

### PR DESCRIPTION
**Pull request by russ**
* Configurable to be on/off. Off by default.
* Configurable to set the maximum request size. Default is 1MB.

## Purpose
A configurable handler for setting a maximum size on requests.

## Description
Fixes #1143

## Checklist
* [X] - An issue already exists detailing the issue/or feature request that this PR fixes
* [X] - All specs are formatted with `crystal tool format spec src`
* [ ] - Inline documentation has been added and/or updated
* [X] - Lucky builds on docker with `./script/setup`
* [X] - All builds and specs pass on docker with `./script/test`

### CODE REVIEW

**Comment by russ** [comment]
Where should docs for this go?

File: spec/lucky/maximum_request_size_handler_spec.cr [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      end
+      context.response.status.should eq(HTTP::Status::OK)
+    end
+
+    it "with a large request, deny the request" do
+      context = build_large_request_context("/path")
+      Lucky::MaximumRequestSizeHandler.temp_config(enabled: true) do
+        run_request_size_handler(context)
+      end
+      context.response.status.should eq(HTTP::Status::PAYLOAD_TOO_LARGE)
```
Comment (jwoertink) [inline spec/lucky/maximum_request_size_handler_spec.cr:31]: seems this failed?

File: spec/spec_helper.cr [hunk 2]
```diff
@@ -48,4 +48,8 @@ Lucky::ForceSSLHandler.configure do |settings|
   settings.enabled = true
 end
 
+Lucky::MaximumRequestSizeHandler.configure do |settings|
+  settings.enabled = false
+end
```
Comment (jwoertink) [inline spec/spec_helper.cr:53]: this is probably not needed since the default is `false`, right?

File: src/lucky/maximum_request_size_handler.cr [hunk 3]
```diff
@@ -0,0 +1,40 @@
+class Lucky::MaximumRequestSizeHandler
+  include HTTP::Handler
+
+  Habitat.create do
+    setting enabled : Bool = false
+    setting max_size : Int32 = 1_048_576 # 1MB
```
Comment (jwoertink) [inline src/lucky/maximum_request_size_handler.cr:6]: :+1:

**Review by jwoertink — COMMENTED** [review]
From what I can tell, this seems like it's fine. I'm not sure why the specs are failing on it though. As for docs, either at the top of the class file just above the class definition, or we just write them up on the website.

File: src/lucky/maximum_request_size_handler.cr (hunk 3 above)
Comment (jwoertink) [reply src/lucky/maximum_request_size_handler.cr:6]: actually... if you wanted this to be say 10GB (video uploads), would this need to be Int64?

**Review by jwoertink — APPROVED** [review]
I think the failure is related to changes in ExceptionPage and probably unrelated. The rest looks good! Thanks :raised_hands:

**Pull request merged**